src/test/java/com/solvd/movie/persistence/EsMovieRepositoryIT.java

```diff
@@ -0,0 +1,114 @@
+package com.solvd.movie.persistence;
+
+import com.solvd.movie.model.EsMovie;
+import com.solvd.movie.model.criteria.SearchCriteria;
+import com.solvd.movie.service.impl.MovieFactory;
+import integration.EsContainer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.util.List;
+
+@SpringBootTest
+@Testcontainers
+public class EsMovieRepositoryIT {
+
+    @Container
+    private static final ElasticsearchContainer container = new EsContainer();
+
+    @DynamicPropertySource
+    private static void properties(final DynamicPropertyRegistry registry) {
+        registry.add("spring.elasticsearch.uris", container::getHttpHostAddress);
+        registry.add("kafka.bootstrap-servers", () -> "localhost");
+    }
+
+    @Autowired
+    private EsMovieRepository movieRepository;
+
+    @BeforeAll
+    public static void start() {
+        container.start();
+    }
+
+    @Test
+    public void verifyFindAll() {
+        EsMovie movie = MovieFactory.getEsMovie();
+        Pageable pageable = PageRequest.of(0, 20);
+        Mono<EsMovie> createdMovie = this.movieRepository.save(movie);
+        StepVerifier.create(createdMovie)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+        Flux<EsMovie> movies = this.movieRepository.findAllByCriteria(
+                new SearchCriteria(), pageable
+        );
+        StepVerifier.create(movies)
+                .expectNext(movie)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    public void verifyFindByCriteria() {
+        SearchCriteria criteria = new SearchCriteria();
+        criteria.setYearFrom(2000);
```

Comment:
@hizmailovich the more fields you will have the more uglier `SearchCriteria` class will be, remove setters and use the power of decorators